app/src/main/java/org/kiwix/kiwixmobile/zim_manager/fileselect_view/effects/OpenFileWithNavigation.kt

```diff
@@ -35,7 +35,7 @@ data class OpenFileWithNavigation(private val bookOnDisk: BooksOnDiskListItem.Bo
     if (!file.canRead()) {
       activity.toast(R.string.error_file_not_found)
     } else {
-      activity.navigate(
```

Comment:
we could have kept the casting in the extension function just so the code isn't filled with all these ugly casts

Reply:
You mean keeping the extension function for AppCompatActivity/Activty and then casting "this" to CoreMainActivity inside the extension function?

Reply:
yeah, it keeps our files less messy and at least centralises the casting